comp/metadata/runner/runnerimpl/runner.go

```diff
@@ -137,12 +176,21 @@ func (r *runnerImpl) handleProvider(p MetadataProvider) {
 // start is called by FX when the application starts. Lifecycle hooks are blocking and called sequencially. We should
 // not block here.
 func (r *runnerImpl) start() error {
-	r.log.Debugf("Starting metadata runner with %d providers", len(r.providers))
-	for _, optionaP := range r.providers {
-		if p, isSet := optionaP.Get(); isSet {
-			go r.handleProvider(p)
+	r.log.Debugf("Starting metadata runner with %d providers", len(r.providers)+len(r.priorityProviders))
+
+	go func() {
+		for _, priorityProvider := range r.priorityProviders {
+			// Execute synchronously the priority provider
+			priorityProvider(context.Background())
```

Comment:
Here is the code that ensure the HTTP request with agent V5 payload is sent first to the intake endpoint